apps/asap-server/test/handlers/teams/index-users-by-team-handler.test.ts

```diff
@@ -0,0 +1,131 @@
+import Boom from '@hapi/boom';
+import { ListUserResponse } from '@asap-hub/model';
+import { BatchRequest } from '@asap-hub/algolia';
+import { indexTeamsUsersHandler } from '../../../src/handlers/teams/index-users-by-team-handler';
+import {
+  getUserResponse,
+  getListUserResponse,
+} from '../../fixtures/users.fixtures';
+
+import {
+  getTeamResponse,
+  createEvent,
+  updateEvent,
+  unpublishedEvent,
+  deleteEvent,
+  TeamEventGenerator,
+} from '../../fixtures/teams.fixtures';
+
+import { algoliaSearchClientMock } from '../../mocks/algolia-client.mock';
+import { userControllerMock } from '../../mocks/user-controller.mock';
+import { teamControllerMock } from '../../mocks/team-controller.mock';
+
+const getUsersBatchCall = (users: ListUserResponse): BatchRequest[] =>
+  users.items.map((user) => ({
+    action: 'updateObject',
+    body: user,
+  }));
+
+const possibleEvents: [string, TeamEventGenerator][] = [
+  ['created', createEvent],
+  ['updated', updateEvent],
+  ['unpublished', unpublishedEvent],
+  ['deleted', deleteEvent],
+];
+
+const possibleRacingConditionEvents: [
+  string,
+  TeamEventGenerator,
+  TeamEventGenerator,
+][] = [
+  ['created and updated', createEvent, updateEvent],
+  ['updated and created', updateEvent, createEvent],
+
+  ['created and unpublished', createEvent, unpublishedEvent],
+  ['unpublished and created', unpublishedEvent, createEvent],
+
+  ['created and deleted', createEvent, deleteEvent],
+  ['deleted and created', deleteEvent, createEvent],
+
+  ['updated and deleted', updateEvent, deleteEvent],
+  ['deleted and updated', deleteEvent, updateEvent],
+
+  ['updated and unpublished', updateEvent, unpublishedEvent],
+  ['unpublished and updated', unpublishedEvent, updateEvent],
+];
+
+describe('Index Users on Team event handler', () => {
```

Comment:
with the concurrent tests i no longer know whats going on so i need to ask - is there a test that shows what happens when only some of the `Promise.allSettled` promises actually settle?